src/QueryNode/QueryNode.jsx

```diff
@@ -188,6 +194,16 @@ class QueryNode extends React.Component {
     return popup;
   }
 
+  userHasDeleteOnProject = () => {
```

Comment:
I would consider move all these kinds of premission checking functions into a single place like `utils` and import from there maybe? So some of the logic could be shared. But just my 2c

Reply:
to be addressed in separate PR since this one tested by QA